ts/nni_manager/training_service/reusable/environments/openPaiEnvironmentService.ts

```diff
@@ -32,9 +31,9 @@ export class OpenPaiEnvironmentService extends EnvironmentService {
     private experimentId: string;
     private config: FlattenOpenpaiConfig;
 
-    constructor(config: ExperimentConfig) {
+    constructor(_experimentRootDir: string, experimentId: string, config: ExperimentConfig) {
```

Comment:
Why pass rootDir and expId from constructor function?

Reply:
Because they are stored in "module scope objects" and custom training services cannot access them.
We must either (1) provide stateful API function to get experiment ID, (2) pass experiment ID through parameter.
According to the meeting last week, (2) is preferred by most people.